ggml-vulkan.cpp

```diff
@@ -2055,7 +2094,7 @@ static vk_pipeline* ggml_vk_guess_matmul_pipeline(ggml_backend_vk_context * ctx,
         return aligned ? &ctx->pipeline_matmul_f16_aligned_l : &ctx->pipeline_matmul_f16_l;
```

Comment:
If you want to detangle this function completely, technically the `subgroupsize == 64` check is a check for AMD, so I guess that could go into a separate function as well.

Reply:
Can we check for `vendor_id == VK_VENDOR_ID_AMD` there or is it better to keep checking for `subgroupsize == 64`?

Reply:
I think it's fine to check for AMD there, for now. I'm not aware of other GPUs with warps of size 64.

Reply:
Done too.